cice5: shared libraries not found

```
$ ldd opt/spack/linux-rocky8-x86_64/intel-2019.5.281/cice5-master-byagur66qd4y7za7vauskh5w34t3cb54/bin/cice_auscom_1440x1080_48x40_480p.exe | grep not
	libnetcdf.so.18 => not found
	libnetcdff.so.7 => not found
```

**Notes**

@penguian made a good suggestion to explicitly run ld while altering the library search directories.

Original:
``
ld     [...] -L/apps/netcdf/4.7.4p/lib [...] -o cice_auscom_360x300_24p.ld-test.exe [...] -L/apps/netcdf/4.7.4p/lib/ompi3/Intel -L/apps/netcdf/4.7.4p/lib/ompi3 -L/apps/netcdf/4.7.4p/lib [...] -L/apps/netcdf/4.7.4p/lib/ompi3/Intel/ -L/apps/netcdf/4.7.4p/lib/ompi3/GNU/ -L/apps/netcdf/4.7.4p/lib/ompi3/ -L/apps/netcdf/4.7.4p/lib/ [...] -lnetcdf -lnetcdff 
``
=> Creates an executable

Remove: -L/apps/netcdf/4.7.4p/lib/ompi3/Intel/ -L/apps/netcdf/4.7.4p/lib/ompi3/GNU/ -L/apps/netcdf/4.7.4p/lib/ompi3/
``
ld    [...] -L/apps/netcdf/4.7.4p/lib [...] -o cice_auscom_360x300_24p.ld-test.exe [...]  -L/apps/netcdf/4.7.4p/lib [...] -lnetcdf -lnetcdff
..
/bin/ld: cannot find -lnetcdf
/bin/ld: cannot find -lnetcdff
``
=> Does not create an executable

Remove: -L/apps/netcdf/4.7.4p/lib/ 
``
ld    [...] -o cice_auscom_360x300_24p.ld-test.exe [...] -L/apps/netcdf/4.7.4p/lib/ompi3/Intel -L/apps/netcdf/4.7.4p/lib/ompi3 [...] -L/apps/netcdf/4.7.4p/lib/ompi3/Intel/ -L/apps/netcdf/4.7.4p/lib/ompi3/GNU/ -L/apps/netcdf/4.7.4p/lib/ompi3/ [...] -lnetcdf -lnetcdff 
``
=> Creates an executable

--- a/packages/cice5/package.py
+++ b/packages/cice5/package.py
@@ -63,6 +63,24 @@ class Cice5(MakefilePackage):
     def url_for_version(self, version):
         return "https://github.com/ACCESS-NRI/cice5/tarball/{0}".format(version)
 
+    # The reason for the explicit -rpath is:
+    # https://github.com/ACCESS-NRI/spack_packages/issues/14#issuecomment-1653651447
+    def get_linker_args(self, spec, name):
+        return " ".join(
+                    [(spec[name].libs).ld_flags,
+                    "-Wl,-rpath=" + join_path(spec[name].prefix, "lib")]
+                   )
+
+    # The reason for the explicit -rpath is:
+    # https://github.com/ACCESS-NRI/spack_packages/issues/14#issuecomment-1653651447
+    def make_linker_args(self, spec, name, namespecs):
+        path = join_path(spec[name].prefix, "lib")
+        return " ".join(
+                    ["-L" + path,
+                    namespecs,
+                    "-Wl,-rpath=" + path]
+                   )
+
     def edit(self, spec, prefix):
 
         srcdir = self.stage.source_path
@@ -77,15 +95,11 @@ class Cice5(MakefilePackage):
         ideps = ["parallelio", "oasis3-mct", "libaccessom2", "netcdf-fortran"]
         incs = " ".join([istr] + [(spec[d].headers).cpp_flags for d in ideps])
 
-        lstr = " ".join(
-                    ["-L" + join_path(spec["parallelio"].prefix, "lib"),
-                    "-lpiof",
-                    "-lpioc"]
-                   )
+        lstr = self.make_linker_args(spec, "parallelio", "-lpiof -lpioc")
         # NOTE: The order of the libraries matter during the linking step!
         # NOTE: datetime-fortran is a dependency of libaccessom2.
         ldeps = ["oasis3-mct", "libaccessom2", "netcdf-c", "netcdf-fortran", "datetime-fortran"]
-        libs = " ".join([lstr] + [(spec[d].libs).ld_flags for d in ldeps])
+        libs = " ".join([lstr] + [self.get_linker_args(spec, d) for d in ldeps])
 
         # Copied from bld/Macros.nci
         config["pre"] = f"""
@@ -184,12 +198,10 @@ ifeq ($(OASIS3_MCT), yes)
    CPPDEFS := $(CPPDEFS) -DOASIS3_MCT
 endif
 """
+        fullconfig = config["pre"] + config[self.compiler.name] + config["post"]
+        print(fullconfig)
         with open(makeinc_path, "w") as makeinc:
-            makeinc.write(
-                config["pre"]
-                + config[self.compiler.name]
-                + config["post"]
-            )
+            makeinc.write(fullconfig)
 
     def build(self, spec, prefix):
 

